operator/apis/loki/v1beta1/alertingrule_defaulting_webhook.go

```diff
@@ -0,0 +1,27 @@
+package v1beta1
+
+import (
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const tenantIDLabelname = "tenantId"
+
+var _ webhook.Defaulter = &AlertingRule{}
+
+func (ar *AlertingRule) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(ar).
+		Complete()
+}
+
+func (ar *AlertingRule) Default() {
+	for _, group := range ar.Spec.Groups {
```

Comment:
Wouldn't it be simpler to do this step when converting the AlertingRule CRs to the rule configuration in [`RulesConfigMap`](https://github.com/grafana/loki/blob/fd627f2a49d94b29091b12e461acb58335363d9e/operator/internal/manifests/rules_config.go#L13)? If this is not depending on any configuration at all, we could see it "as part of the marshalling" and even put it in [`MarshalAlertingRule`](https://github.com/grafana/loki/blob/fd627f2a49d94b29091b12e461acb58335363d9e/operator/internal/manifests/internal/rules/marshal.go#L18).

This would make it easy to write a test for it and we would not need the webhook at all.